### PR TITLE
Updated Release Notes for R2.1.

### PR DESCRIPTION
**Pull request by jrobble**
This change is [Reviewable](https://reviewable.io/reviews/openmpf/openmpf.github.io/59)

### CODE REVIEW

**Comment by jrobble** [comment]
Note that this PR still has TODOs that need to be addressed prior to finalizing R2.1. Some of them are pending the completion of other open PRs.

**Comment by clnowacki** [comment]
Review status: 0 of 5 files reviewed, 15 unresolved discussions (waiting on @jrobble and @clnowacki)

---

*[docs/docs/Release-Notes.md, line 30 at r2](https://reviewable.io/reviews/openmpf/openmpf.github.io/59#-LEt8gh29M-eD9Spu33i:-LEt8gh29M-eD9Spu33j:b-nqzpvi) ([raw file](https://github.com/openmpf/openmpf.github.io/blob/3660cf7d5ccc4454b9effa774a17defa1e563533/docs/docs/Release-Notes.md#L30)):*
> ```Markdown
> - Note that changes made using the Component Registration web page only affect core nodes, not spare nodes. Core nodes are those configured during the OpenMPF installation process.
> - Added `mpf list-nodes` command to list the core nodes and available spare nodes.
> - Now using JGroups FILE_PING protocol for peer discovery instead of TCPPING. This means that the list of OpenMPF nodes no longer needs to be fully specified when the Workflow Manager starts. Instead, the files in `$MPF_HOME/share/nodes` are used by the Node Manager process on each node, and the Workflow Manager, to determine which nodes are currently available.
> ```

This first sentence is a fragment. Change it to "the MPF system now uses..." or something like that.

I think that last sentence will be clearer if it does not use the passive voice: "Instead, the Workflow Manager and Node Manager process on each node use the files in `$MPF_HOME/share/nodes` to determine which nodes are currently available.

---

*[docs/docs/Release-Notes.md, line 32 at r2](https://reviewable.io/reviews/openmpf/openmpf.github.io/59#-LEt9sLsAHDcaJ57gMxM:-LEt9sLsAHDcaJ57gMxN:bpi7p9y) ([raw file](https://github.com/openmpf/openmpf.github.io/blob/3660cf7d5ccc4454b9effa774a17defa1e563533/docs/docs/Release-Notes.md#L32)):*
> ```Markdown
> - Now using JGroups FILE_PING protocol for peer discovery instead of TCPPING. This means that the list of OpenMPF nodes no longer needs to be fully specified when the Workflow Manager starts. Instead, the files in `$MPF_HOME/share/nodes` are used by the Node Manager process on each node, and the Workflow Manager, to determine which nodes are currently available.
> - Updated JGroups from 3.6.4. to 4.0.11.
> - The environment variables specified in `/etc/profile.d/mpf.sh` have been simplified.
> ```

If the user is upgrading an existing system, will the new mpf.sh file be automatically merged into the existing one for them, or does the user need to know to do that themselves?

---

*[docs/docs/Release-Notes.md, line 36 at r2](https://reviewable.io/reviews/openmpf/openmpf.github.io/59#-LEtAGPrCWWb1e0UFEKh:-LEtAGPrCWWb1e0UFEKi:bnl8bpb) ([raw file](https://github.com/openmpf/openmpf.github.io/blob/3660cf7d5ccc4454b9effa774a17defa1e563533/docs/docs/Release-Notes.md#L36)):*
> ```Markdown
> <h2>Default Detection System Properties</h2>
> 
> - Some system properties can now be updated at runtime without restarting the Workflow Manager. Specifically, they are the properties that specify the default values when creating new jobs. Changing these properties will only have an effect on new jobs, not jobs that are currently running.
> ```

That first sentence should be more specific about which properties can be updated: "... they are the properties that specify default values for detection algorithms when creating new jobs."

---

*[docs/docs/Release-Notes.md, line 37 at r2](https://reviewable.io/reviews/openmpf/openmpf.github.io/59#-LEtAj6N-qbXesVn8jzN:-LEtAj6N-qbXesVn8jzO:bq38nwm) ([raw file](https://github.com/openmpf/openmpf.github.io/blob/3660cf7d5ccc4454b9effa774a17defa1e563533/docs/docs/Release-Notes.md#L37)):*
> ```Markdown
> 
> - Some system properties can now be updated at runtime without restarting the Workflow Manager. Specifically, they are the properties that specify the default values when creating new jobs. Changing these properties will only have an effect on new jobs, not jobs that are currently running.
> - These default detection system properties are separated from the general system properties in the Properties web page. The latter still require the Workflow Manager to be restarted to take effect.
> ```

"... to be restarted for changes to take effect."

---

*[docs/docs/Release-Notes.md, line 38 at r2](https://reviewable.io/reviews/openmpf/openmpf.github.io/59#-LEtAs5-8DK-gSgOimK9:-LEtAs5-8DK-gSgOimKA:blgl2j7) ([raw file](https://github.com/openmpf/openmpf.github.io/blob/3660cf7d5ccc4454b9effa774a17defa1e563533/docs/docs/Release-Notes.md#L38)):*
> ```Markdown
> - Some system properties can now be updated at runtime without restarting the Workflow Manager. Specifically, they are the properties that specify the default values when creating new jobs. Changing these properties will only have an effect on new jobs, not jobs that are currently running.
> - These default detection system properties are separated from the general system properties in the Properties web page. The latter still require the Workflow Manager to be restarted to take effect.
> - The Apache Commons Configuration library is now used to read and write properties files. When defining a property value using an environment variable, be sure to prepend the variable name with `env:`. For example:
> ```

When defining a property value using an environment variable <where?> Properties file, pom file, other??

---

*[docs/docs/Release-Notes.md, line 64 at r2](https://reviewable.io/reviews/openmpf/openmpf.github.io/59#-LEtBN0mEmoMDfjS8I2o:-LEtBN0mEmoMDfjS8I2p:b-w96ncj) ([raw file](https://github.com/openmpf/openmpf.github.io/blob/3660cf7d5ccc4454b9effa774a17defa1e563533/docs/docs/Release-Notes.md#L64)):*
> ```Markdown
> <h2>Darknet Component</h2>
> 
> - Implemented a component that uses the [Darknet neural network framework](https://pjreddie.com/darknet/) to perform detection and classification of objects using trained models.
> ```

"This release includes a component ..."

---

*[docs/docs/Release-Notes.md, line 66 at r2](https://reviewable.io/reviews/openmpf/openmpf.github.io/59#-LEtBZHYD1un7LSuxKmw:-LEtBZHYD1un7LSuxKmx:b-7j92ob) ([raw file](https://github.com/openmpf/openmpf.github.io/blob/3660cf7d5ccc4454b9effa774a17defa1e563533/docs/docs/Release-Notes.md#L66)):*
> ```Markdown
> - Implemented a component that uses the [Darknet neural network framework](https://pjreddie.com/darknet/) to perform detection and classification of objects using trained models.
> - Pipelines for the Tiny YOLO and YOLOv2 models are provided. Due to its large size, the YOLOv2 weights file must be downloaded separately and placed in `$MPF_HOME/share/models/DarknetDetection` in order to use the YOLOv2 pipelines. Refer to `DarknetDetection/plugin-files/models/models.ini` for more information.
> - Supports a preprocessor mode and default mode of operation. If preprocessor mode is enabled, and multiple Darknet detections in a frame share the same classification, then those are merged into a single detection where the region corresponds to the superset region that encapsulates all of the original detections, and the confidence value is the probability that at least one of the original detections is a true positive. If disabled, multiple Darknet detections in a frame are not merged together.
> ```

"This component supports a preprocessor mode..."

---

*[docs/docs/Release-Notes.md, line 68 at r2](https://reviewable.io/reviews/openmpf/openmpf.github.io/59#-LEtBlTf7E1oDoDo1OTH:-LEtBlTf7E1oDoDo1OTI:b-a431qx) ([raw file](https://github.com/openmpf/openmpf.github.io/blob/3660cf7d5ccc4454b9effa774a17defa1e563533/docs/docs/Release-Notes.md#L68)):*
> ```Markdown
> - Supports a preprocessor mode and default mode of operation. If preprocessor mode is enabled, and multiple Darknet detections in a frame share the same classification, then those are merged into a single detection where the region corresponds to the superset region that encapsulates all of the original detections, and the confidence value is the probability that at least one of the original detections is a true positive. If disabled, multiple Darknet detections in a frame are not merged together.
> - Detections are not tracked across frames. One track is generated per detection.
> - Supports an optional CLASS_WHITELIST_FILE property. When provided, only detections with class names listed in the file will be generated.
> ```

"The component supports..."

---

*[docs/docs/Release-Notes.md, line 69 at r2](https://reviewable.io/reviews/openmpf/openmpf.github.io/59#-LEtBqshBRTZfL1QnphZ:-LEtBqshBRTZfL1Qnph_:b28iuub) ([raw file](https://github.com/openmpf/openmpf.github.io/blob/3660cf7d5ccc4454b9effa774a17defa1e563533/docs/docs/Release-Notes.md#L69)):*
> ```Markdown
> - Detections are not tracked across frames. One track is generated per detection.
> - Supports an optional CLASS_WHITELIST_FILE property. When provided, only detections with class names listed in the file will be generated.
> - Can be compiled with GPU support if the NVIDIA CUDA Toolkit is installed on the build machine. Refer to the [GPU Support Guide](GPU-Support-Guide). If the toolkit is not found, then the component will compile with CPU support only.
> ```

"The component can be compiled..."

---

*[docs/docs/Release-Notes.md, line 70 at r2](https://reviewable.io/reviews/openmpf/openmpf.github.io/59#-LEtBwtaEUpLUMaxFaSO:-LEtBwtb6sGd-ptVzfce:b-23wo0y) ([raw file](https://github.com/openmpf/openmpf.github.io/blob/3660cf7d5ccc4454b9effa774a17defa1e563533/docs/docs/Release-Notes.md#L70)):*
> ```Markdown
> - Supports an optional CLASS_WHITELIST_FILE property. When provided, only detections with class names listed in the file will be generated.
> - Can be compiled with GPU support if the NVIDIA CUDA Toolkit is installed on the build machine. Refer to the [GPU Support Guide](GPU-Support-Guide). If the toolkit is not found, then the component will compile with CPU support only.
> - To run on a GPU, set CUDA_DEVICE_ID >= 0, or set the detection.cuda.device.id system property.
> ```

Set CUDA_DEVICE_ID>=0 where?

---

*[docs/docs/Release-Notes.md, line 71 at r2](https://reviewable.io/reviews/openmpf/openmpf.github.io/59#-LEtCCga5gTMP37hlpir:-LEtCCga5gTMP37hlpis:bs7h6ks) ([raw file](https://github.com/openmpf/openmpf.github.io/blob/3660cf7d5ccc4454b9effa774a17defa1e563533/docs/docs/Release-Notes.md#L71)):*
> ```Markdown
> - Can be compiled with GPU support if the NVIDIA CUDA Toolkit is installed on the build machine. Refer to the [GPU Support Guide](GPU-Support-Guide). If the toolkit is not found, then the component will compile with CPU support only.
> - To run on a GPU, set CUDA_DEVICE_ID >= 0, or set the detection.cuda.device.id system property.
> - When CUDA_DEVICE_ID >= 0, set FALLBACK_TO_CPU_WHEN_GPU_PROBLEM, or the detection.use.cpu.when.gpu.problem system property, to true to run the component logic on the CPU instead of the GPU when a GPU problem is detected.
> ```

Can you put the word 'true' in quotes or something so that it stands out?

Also, I would reword this as "When CUDA_DEVICE_ID >= 0, you can set FALLBACK_TO_CPU_WHEN_GPU_PROBLEM, or the detection.use.cpu.when.gpu.problem system property, to "true"  if you want to run the component logic on the CPU instead of the GPU when a GPU problem is detected."

---

*[docs/docs/Release-Notes.md, line 75 at r2](https://reviewable.io/reviews/openmpf/openmpf.github.io/59#-LEtCyBsAsM2q61Qx4rG:-LEtCyBsAsM2q61Qx4rH:b-5vsj3m) ([raw file](https://github.com/openmpf/openmpf.github.io/blob/3660cf7d5ccc4454b9effa774a17defa1e563533/docs/docs/Release-Notes.md#L75)):*
> ```Markdown
> <h2>Models Directory</h2>
> 
> - The`$MPF_HOME/share/models` directory is now used by the Darknet and Caffe components to store model files and associated files, such as classification names files, weights files, etc. This allows users to more easily add model files post-deployment. Instead of copying the model files to `$MPF_HOME/plugins/<component-name>/models` directory on each node in the OpenMPF cluster, they only need to be placed in the shared directory once.
> ```

"... they only need to copy them to the shared directory once."

---

*[docs/docs/Release-Notes.md, line 91 at r2](https://reviewable.io/reviews/openmpf/openmpf.github.io/59#-LEtDJPF69R-mxIjOhVV:-LEtDJPF69R-mxIjOhVW:b-bg96j0) ([raw file](https://github.com/openmpf/openmpf.github.io/blob/3660cf7d5ccc4454b9effa774a17defa1e563533/docs/docs/Release-Notes.md#L91)):*
> ```Markdown
> <h2>Bug Fixes</h2>
> 
> - Custom Action, task, and pipeline names can now contain "(" and ")" characters, again.
> ```

Remove the comma between "characters" and "again".

---

*[docs/docs/Release-Notes.md, line 92 at r2](https://reviewable.io/reviews/openmpf/openmpf.github.io/59#-LEtDUK13fNR7fkn2Z8t:-LEtDUK2-MWT32-7StIc:bnded6g) ([raw file](https://github.com/openmpf/openmpf.github.io/blob/3660cf7d5ccc4454b9effa774a17defa1e563533/docs/docs/Release-Notes.md#L92)):*
> ```Markdown
> 
> - Custom Action, task, and pipeline names can now contain "(" and ")" characters, again.
> - Detection elements for audio tracks and generic tracks in a JSON output object will now have a y value of 0 instead of 1.
> ```

"Detection location elements"

---

*[docs/docs/Release-Notes.md, line 95 at r2](https://reviewable.io/reviews/openmpf/openmpf.github.io/59#-LEtDg717vD3u6CUfY1D:-LEtDg717vD3u6CUfY1E:b-8peubh) ([raw file](https://github.com/openmpf/openmpf.github.io/blob/3660cf7d5ccc4454b9effa774a17defa1e563533/docs/docs/Release-Notes.md#L95)):*
> ```Markdown
> - Streaming health report and summary report timestamps have been corrected to represent hours in the 1-24 range instead of 0-23.
> - Single-frame .gif files are now segmented properly and no longer result in a NullPointerException. 
> - Now setting LD_LIBRARY_PATH at the process level for Tomcat, the Node Manager, and component services, instead of at the system level in `/etc/profile.d/mpf.sh`. Also, deployments no longer create `/etc/ld.so.conf.d/mpf.conf`. This better isolates OpenMPF from the rest of the system and prevents issues, such as being unable to use SSH, when system libraries are not compatible with OpenMPF libraries. The latter situation may occur when running "yum update" on the system, which can make OpenMPF unusable until a new deployment package with compatible libraries is installed. 
> ```

"LD_LIBRARY_PATH is now set..."

---


*Comments from [Reviewable](https://reviewable.io/reviews/openmpf/openmpf.github.io/59)*

**Comment by jrobble** [comment]
Review status: 0 of 5 files reviewed, 15 unresolved discussions (waiting on @clnowacki and @jrobble)

---

*[docs/docs/Release-Notes.md, line 30 at r2](https://reviewable.io/reviews/openmpf/openmpf.github.io/59#-LEt8gh29M-eD9Spu33i:-LEtX_qi8ZJ2zEDq4_R3:b-896fix) ([raw file](https://github.com/openmpf/openmpf.github.io/blob/3660cf7d5ccc4454b9effa774a17defa1e563533/docs/docs/Release-Notes.md#L30)):*
<details><summary><i>Previously, clnowacki wrote…</i></summary><blockquote>

This first sentence is a fragment. Change it to "the MPF system now uses..." or something like that.

I think that last sentence will be clearer if it does not use the passive voice: "Instead, the Workflow Manager and Node Manager process on each node use the files in `$MPF_HOME/share/nodes` to determine which nodes are currently available.
</blockquote></details>

Done.

---

*[docs/docs/Release-Notes.md, line 32 at r2](https://reviewable.io/reviews/openmpf/openmpf.github.io/59#-LEt9sLsAHDcaJ57gMxM:-LEtXdOTB--qNJ5lGARn:bq3optq) ([raw file](https://github.com/openmpf/openmpf.github.io/blob/3660cf7d5ccc4454b9effa774a17defa1e563533/docs/docs/Release-Notes.md#L32)):*
<details><summary><i>Previously, clnowacki wrote…</i></summary><blockquote>

If the user is upgrading an existing system, will the new mpf.sh file be automatically merged into the existing one for them, or does the user need to know to do that themselves?
</blockquote></details>

During an upgrade, the existing mpf.sh file is replaced with a new one. The user should not have to do anything. The only time a user should modify the contents of the mpf.sh file is when configuring a spare node after the cloning process, as discussed in the new Node Guide. I plan to test the upgrade procedure today or tomorrow.

---

*[docs/docs/Release-Notes.md, line 36 at r2](https://reviewable.io/reviews/openmpf/openmpf.github.io/59#-LEtAGPrCWWb1e0UFEKh:-LEtZ3BI9YA9gPSfq9wE:b-896fix) ([raw file](https://github.com/openmpf/openmpf.github.io/blob/3660cf7d5ccc4454b9effa774a17defa1e563533/docs/docs/Release-Notes.md#L36)):*
<details><summary><i>Previously, clnowacki wrote…</i></summary><blockquote>

That first sentence should be more specific about which properties can be updated: "... they are the properties that specify default values for detection algorithms when creating new jobs."
</blockquote></details>

Done.

---

*[docs/docs/Release-Notes.md, line 37 at r2](https://reviewable.io/reviews/openmpf/openmpf.github.io/59#-LEtAj6N-qbXesVn8jzN:-LEtZKBD1lCOPQd3WJOL:b-896fix) ([raw file](https://github.com/openmpf/openmpf.github.io/blob/3660cf7d5ccc4454b9effa774a17defa1e563533/docs/docs/Release-Notes.md#L37)):*
<details><summary><i>Previously, clnowacki wrote…</i></summary><blockquote>

"... to be restarted for changes to take effect."
</blockquote></details>

Done.

---

*[docs/docs/Release-Notes.md, line 38 at r2](https://reviewable.io/reviews/openmpf/openmpf.github.io/59#-LEtAs5-8DK-gSgOimK9:-LEtZy1O-cF4T0kRYnpV:b-896fix) ([raw file](https://github.com/openmpf/openmpf.github.io/blob/3660cf7d5ccc4454b9effa774a17defa1e563533/docs/docs/Release-Notes.md#L38)):*
<details><summary><i>Previously, clnowacki wrote…</i></summary><blockquote>

When defining a property value using an environment variable <where?> Properties file, pom file, other??
</blockquote></details>

Done.

---

*[docs/docs/Release-Notes.md, line 64 at r2](https://reviewable.io/reviews/openmpf/openmpf.github.io/59#-LEtBN0mEmoMDfjS8I2o:-LEt_7ibEV7qjr08j_GC:b-896fix) ([raw file](https://github.com/openmpf/openmpf.github.io/blob/3660cf7d5ccc4454b9effa774a17defa1e563533/docs/docs/Release-Notes.md#L64)):*
<details><summary><i>Previously, clnowacki wrote…</i></summary><blockquote>

"This release includes a component ..."
</blockquote></details>

Done.

---

*[docs/docs/Release-Notes.md, line 66 at r2](https://reviewable.io/reviews/openmpf/openmpf.github.io/59#-LEtBZHYD1un7LSuxKmw:-LEt_EnxDLfCL81XHyku:b-896fix) ([raw file](https://github.com/openmpf/openmpf.github.io/blob/3660cf7d5ccc4454b9effa774a17defa1e563533/docs/docs/Release-Notes.md#L66)):*
<details><summary><i>Previously, clnowacki wrote…</i></summary><blockquote>

"This component supports a preprocessor mode..."
</blockquote></details>

Done.

---

*[docs/docs/Release-Notes.md, line 68 at r2](https://reviewable.io/reviews/openmpf/openmpf.github.io/59#-LEtBlTf7E1oDoDo1OTH:-LEt_Hmd1LvVkIotAKM1:b-896fix) ([raw file](https://github.com/openmpf/openmpf.github.io/blob/3660cf7d5ccc4454b9effa774a17defa1e563533/docs/docs/Release-Notes.md#L68)):*
<details><summary><i>Previously, clnowacki wrote…</i></summary><blockquote>

"The component supports..."
</blockquote></details>

Done.

---

*[docs/docs/Release-Notes.md, line 69 at r2](https://reviewable.io/reviews/openmpf/openmpf.github.io/59#-LEtBqshBRTZfL1QnphZ:-LEt_RlbDc8MlivyC75n:b-896fix) ([raw file](https://github.com/openmpf/openmpf.github.io/blob/3660cf7d5ccc4454b9effa774a17defa1e563533/docs/docs/Release-Notes.md#L69)):*
<details><summary><i>Previously, clnowacki wrote…</i></summary><blockquote>

"The component can be compiled..."
</blockquote></details>

Done.

---

*[docs/docs/Release-Notes.md, line 70 at r2](https://reviewable.io/reviews/openmpf/openmpf.github.io/59#-LEtBwtaEUpLUMaxFaSO:-LEtaY8G3o_OwI189UH5:b-896fix) ([raw file](https://github.com/openmpf/openmpf.github.io/blob/3660cf7d5ccc4454b9effa774a17defa1e563533/docs/docs/Release-Notes.md#L70)):*
<details><summary><i>Previously, clnowacki wrote…</i></summary><blockquote>

Set CUDA_DEVICE_ID>=0 where?
</blockquote></details>

Done.

---

*[docs/docs/Release-Notes.md, line 71 at r2](https://reviewable.io/reviews/openmpf/openmpf.github.io/59#-LEtCCga5gTMP37hlpir:-LEtaYe-7QGkJjsdSIpz:b-896fix) ([raw file](https://github.com/openmpf/openmpf.github.io/blob/3660cf7d5ccc4454b9effa774a17defa1e563533/docs/docs/Release-Notes.md#L71)):*
<details><summary><i>Previously, clnowacki wrote…</i></summary><blockquote>

Can you put the word 'true' in quotes or something so that it stands out?

Also, I would reword this as "When CUDA_DEVICE_ID >= 0, you can set FALLBACK_TO_CPU_WHEN_GPU_PROBLEM, or the detection.use.cpu.when.gpu.problem system property, to "true"  if you want to run the component logic on the CPU instead of the GPU when a GPU problem is detected."
</blockquote></details>

Done.

---

*[docs/docs/Release-Notes.md, line 75 at r2](https://reviewable.io/reviews/openmpf/openmpf.github.io/59#-LEtCyBsAsM2q61Qx4rG:-LEtal3M7ZRm5wWAuG2E:b-896fix) ([raw file](https://github.com/openmpf/openmpf.github.io/blob/3660cf7d5ccc4454b9effa774a17defa1e563533/docs/docs/Release-Notes.md#L75)):*
<details><summary><i>Previously, clnowacki wrote…</i></summary><blockquote>

"... they only need to copy them to the shared directory once."
</blockquote></details>

Done.

---

*[docs/docs/Release-Notes.md, line 91 at r2](https://reviewable.io/reviews/openmpf/openmpf.github.io/59#-LEtDJPF69R-mxIjOhVV:-LEtauHY0GQMtYAZwDQR:b-896fix) ([raw file](https://github.com/openmpf/openmpf.github.io/blob/3660cf7d5ccc4454b9effa774a17defa1e563533/docs/docs/Release-Notes.md#L91)):*
<details><summary><i>Previously, clnowacki wrote…</i></summary><blockquote>

Remove the comma between "characters" and "again".
</blockquote></details>

Done.

---

*[docs/docs/Release-Notes.md, line 92 at r2](https://reviewable.io/reviews/openmpf/openmpf.github.io/59#-LEtDUK13fNR7fkn2Z8t:-LEtaumpCB_lRiKT7hFo:b-896fix) ([raw file](https://github.com/openmpf/openmpf.github.io/blob/3660cf7d5ccc4454b9effa774a17defa1e563533/docs/docs/Release-Notes.md#L92)):*
<details><summary><i>Previously, clnowacki wrote…</i></summary><blockquote>

"Detection location elements"
</blockquote></details>

Done.

---

*[docs/docs/Release-Notes.md, line 95 at r2](https://reviewable.io/reviews/openmpf/openmpf.github.io/59#-LEtDg717vD3u6CUfY1D:-LEtcHTaCkoND9w5VoBT:b-896fix) ([raw file](https://github.com/openmpf/openmpf.github.io/blob/3660cf7d5ccc4454b9effa774a17defa1e563533/docs/docs/Release-Notes.md#L95)):*
<details><summary><i>Previously, clnowacki wrote…</i></summary><blockquote>

"LD_LIBRARY_PATH is now set..."
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io/reviews/openmpf/openmpf.github.io/59)*

**Comment by clnowacki** [comment]
Review status: 0 of 5 files reviewed, 2 unresolved discussions (waiting on @clnowacki)

---

*[docs/docs/Release-Notes.md, line 32 at r2](https://reviewable.io/reviews/openmpf/openmpf.github.io/59#-LEt9sLsAHDcaJ57gMxM:-LEtdhc5AzPIN1RnzbMO:b8tf4fd) ([raw file](https://github.com/openmpf/openmpf.github.io/blob/3660cf7d5ccc4454b9effa774a17defa1e563533/docs/docs/Release-Notes.md#L32)):*
<details><summary><i>Previously, jrobble (Jeff Robble) wrote…</i></summary><blockquote>

During an upgrade, the existing mpf.sh file is replaced with a new one. The user should not have to do anything. The only time a user should modify the contents of the mpf.sh file is when configuring a spare node after the cloning process, as discussed in the new Node Guide. I plan to test the upgrade procedure today or tomorrow.
</blockquote></details>

Well, suppose they are upgrading from 2.1 to 2.2, and they've configured spare nodes in the mpf.sh file. Will the installation process overwrite their changes, or will they be backed up for them?

---

*[docs/docs/Release-Notes.md, line 36 at r2](https://reviewable.io/reviews/openmpf/openmpf.github.io/59#-LEtAGPrCWWb1e0UFEKh:-LEtdx908VzMClHA8vzS:b-djn75y) ([raw file](https://github.com/openmpf/openmpf.github.io/blob/3660cf7d5ccc4454b9effa774a17defa1e563533/docs/docs/Release-Notes.md#L36)):*
<details><summary><i>Previously, jrobble (Jeff Robble) wrote…</i></summary><blockquote>

Done.
</blockquote></details>

Isn't the point that the properties that can be changed is restricted to those having to do with detection?

---


*Comments from [Reviewable](https://reviewable.io/reviews/openmpf/openmpf.github.io/59)*

**Comment by jrobble** [comment]
Reviewed 5 of 5 files at r3.
Review status: all files reviewed, 2 unresolved discussions (waiting on @jrobble and @clnowacki)

---

*[docs/docs/Release-Notes.md, line 32 at r2](https://reviewable.io/reviews/openmpf/openmpf.github.io/59#-LEt9sLsAHDcaJ57gMxM:-LEtjYK43AonHKlk2r31:b-z84h5e) ([raw file](https://github.com/openmpf/openmpf.github.io/blob/3660cf7d5ccc4454b9effa774a17defa1e563533/docs/docs/Release-Notes.md#L32)):*
<details><summary><i>Previously, clnowacki wrote…</i></summary><blockquote>

Well, suppose they are upgrading from 2.1 to 2.2, and they've configured spare nodes in the mpf.sh file. Will the installation process overwrite their changes, or will they be backed up for them?
</blockquote></details>

Only the mpf.sh file on the spare node needs to be updated when adding a spare node to the cluster (otherwise the spare node, and the one it was cloned from, would have this same name). The mpf.sh file on the other nodes does not need to be modified when adding a spare node.

Users should not upgrade a spare node. Instead they should upgrade one of the core child nodes and clone it to create a new, upgraded, spare node. Like Docker containers, spare nodes are intended to be throw-away. It's one step towards the Docker way of doing things.

---

*[docs/docs/Release-Notes.md, line 36 at r2](https://reviewable.io/reviews/openmpf/openmpf.github.io/59#-LEtAGPrCWWb1e0UFEKh:-LEtkRF_9pvEbs0eZXZ1:b-r9j9jz) ([raw file](https://github.com/openmpf/openmpf.github.io/blob/3660cf7d5ccc4454b9effa774a17defa1e563533/docs/docs/Release-Notes.md#L36)):*
Yes, those are the same properties "that specify the default values when creating new jobs". Should I change it to:

> The **detection** properties that specify the default values when creating new jobs can now be updated at runtime without restarting the Workflow Manager.
 

---


*Comments from [Reviewable](https://reviewable.io/reviews/openmpf/openmpf.github.io/59)*

**Comment by clnowacki** [comment]
Review status: all files reviewed, 1 unresolved discussion (waiting on @clnowacki)

---

*[docs/docs/Release-Notes.md, line 32 at r2](https://reviewable.io/reviews/openmpf/openmpf.github.io/59#-LEt9sLsAHDcaJ57gMxM:-LEtlMb01sCoB0UEoypE:bkucfjd) ([raw file](https://github.com/openmpf/openmpf.github.io/blob/3660cf7d5ccc4454b9effa774a17defa1e563533/docs/docs/Release-Notes.md#L32)):*
<details><summary><i>Previously, jrobble (Jeff Robble) wrote…</i></summary><blockquote>

Only the mpf.sh file on the spare node needs to be updated when adding a spare node to the cluster (otherwise the spare node, and the one it was cloned from, would have this same name). The mpf.sh file on the other nodes does not need to be modified when adding a spare node.

Users should not upgrade a spare node. Instead they should upgrade one of the core child nodes and clone it to create a new, upgraded, spare node. Like Docker containers, spare nodes are intended to be throw-away. It's one step towards the Docker way of doing things.


</blockquote></details>

Ok, I get it. Now, suppose the user has been using an environment variable from the mpf.sh file for some other purpose. Should we be more specific about precisely which environment variables were "simplified" and what the nature of the change was?

---

*[docs/docs/Release-Notes.md, line 36 at r2](https://reviewable.io/reviews/openmpf/openmpf.github.io/59#-LEtAGPrCWWb1e0UFEKh:-LEtmIgz5H4xLtNCgHfs:bb77ufg) ([raw file](https://github.com/openmpf/openmpf.github.io/blob/3660cf7d5ccc4454b9effa774a17defa1e563533/docs/docs/Release-Notes.md#L36)):*
<details><summary><i>Previously, jrobble (Jeff Robble) wrote…</i></summary><blockquote>

Yes, those are the same properties "that specify the default values when creating new jobs". Should I change it to:

> The **detection** properties that specify the default values when creating new jobs can now be updated at runtime without restarting the Workflow Manager.
 
</blockquote></details>

Yes.

---


*Comments from [Reviewable](https://reviewable.io/reviews/openmpf/openmpf.github.io/59)*

**Comment by jrobble** [comment]
Review status: 0 of 8 files reviewed, 1 unresolved discussion (waiting on @jrobble and @clnowacki)

---

*[docs/docs/Release-Notes.md, line 32 at r2](https://reviewable.io/reviews/openmpf/openmpf.github.io/59#-LEt9sLsAHDcaJ57gMxM:-LEuG1DHEIC-WIf5358K:b-r29wqf) ([raw file](https://github.com/openmpf/openmpf.github.io/blob/3660cf7d5ccc4454b9effa774a17defa1e563533/docs/docs/Release-Notes.md#L32)):*
<details><summary><i>Previously, clnowacki wrote…</i></summary><blockquote>

Ok, I get it. Now, suppose the user has been using an environment variable from the mpf.sh file for some other purpose. Should we be more specific about precisely which environment variables were "simplified" and what the nature of the change was?
</blockquote></details>

Some environment variables have been added, but the only one that could affect a user is the removal of ALL_MPF_NODES. It has now been replaced with CORE_MPF_NODES. I hope no one is using that for anything, but I will mention it anyway.

---


*Comments from [Reviewable](https://reviewable.io/reviews/openmpf/openmpf.github.io/59)*

**Comment by clnowacki** [comment]
Reviewed 1 of 9 files at r4, 5 of 10 files at r5, 1 of 5 files at r6.
Review status: 7 of 11 files reviewed, 1 unresolved discussion (waiting on @jrobble)

---

*[docs/docs/Release-Notes.md, line 108 at r6](https://reviewable.io/reviews/openmpf/openmpf.github.io/59#-LFX_eZ7035ODXJdHj7-:-LFX_eZ7035ODXJdHj70:b-ooi17k) ([raw file](https://github.com/openmpf/openmpf.github.io/blob/0ef3d89a372671447a1f036e058fe479d3b6551d/docs/docs/Release-Notes.md#L108)):*
> ```Markdown
>    > Further debugging is necessary to determine the reason why that message was missing that field. The situation is not easily reproducible. It may occur when ActiveMQ and / or the system is under heavy load and sends duplicate messages in attempt to ensure message delivery. Some of those messages seem to end up in the dead letter queue (DLQ). For now, we've improved the way we handle messages in the DLQ. If OpenMPF can process a message successfully, the job is marked as COMPLETED_WITH_ERRORS, and the message is moved from ActiveMQ.DLQ to MPF.DLQ_PROCESSED_MESSAGES. If OpenMPF cannot process a message successfully, it is moved from ActiveMQ.DLQ to MPF.DLQ_INVALID_MESSAGES.
> 
> - The `mpf stop` command will stop the Workflow Manager, which will in turn send commands to all of the available nodes to stop all running component services. If a service is processing a sub-job when the quit command is received, that service process will not terminate until that sub-job is completely processed. Thus, the service may put a sub-job response on the ActiveMQ response queue after the Workflow Manager has terminated. That will not cause a problem because the queues are flushed the next time the Workflow Manager starts; however, there will be a problem if the service finishes processing the sub-job after the Workflow Manager is restarted. At that time, the Workflow Manager will have no knowledge of the old job and will in turn generate warnings in the log about how the job id is "not known to the system" and/or "not found as a batch or a streaming job". These can be safely ignored. Often, if these messages appear in the log, then C++ services were running after stopping the Workflow Manager. To address this. you may wish to run `sudo killall amq_detection_component` after running `mpf stop`.
> ```

Sorry if I'm repeating myself: Please change the period to a comma after "To address this" in the last sentence.

---


*Comments from [Reviewable](https://reviewable.io/reviews/openmpf/openmpf.github.io/59)*

**Comment by jrobble** [comment]
Review status: 7 of 11 files reviewed, 1 unresolved discussion (waiting on @jrobble and @clnowacki)

---

*[docs/docs/Release-Notes.md, line 108 at r6](https://reviewable.io/reviews/openmpf/openmpf.github.io/59#-LFX_eZ7035ODXJdHj7-:-LFXb2QZ3kJq3cXRO008:b-896fix) ([raw file](https://github.com/openmpf/openmpf.github.io/blob/0ef3d89a372671447a1f036e058fe479d3b6551d/docs/docs/Release-Notes.md#L108)):*
<details><summary><i>Previously, clnowacki wrote…</i></summary><blockquote>

Sorry if I'm repeating myself: Please change the period to a comma after "To address this" in the last sentence.
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io/reviews/openmpf/openmpf.github.io/59)*

**Comment by clnowacki** [comment]
Reviewed 1 of 4 files at r7.
Review status: 7 of 11 files reviewed, 1 unresolved discussion (waiting on @jrobble and @clnowacki)

---



*Comments from [Reviewable](https://reviewable.io/reviews/openmpf/openmpf.github.io/59)*

**Comment by clnowacki** [comment]
:lgtm:

---

Reviewed 1 of 5 files at r6, 3 of 4 files at r7.
Review status: :shipit: complete! all files reviewed, all discussions resolved

---



*Comments from [Reviewable](https://reviewable.io/reviews/openmpf/openmpf.github.io/59#-:-LFXbkf4A27kSBZmAGIP:bnfp4nl)*
